enioka_scan/src/androidTest/java/com/enioka/scanner/sdk/mock/MockTest.java

```diff
@@ -0,0 +1,116 @@
+package com.enioka.scanner.sdk.mock;
+
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.ServiceConnection;
+import android.os.IBinder;
+import android.support.test.InstrumentationRegistry;
+
+import com.enioka.scanner.LaserScanner;
+import com.enioka.scanner.api.Scanner;
+import com.enioka.scanner.api.ScannerConnectionHandler;
+import com.enioka.scanner.api.ScannerSearchOptions;
+import com.enioka.scanner.service.ScannerService;
+import com.enioka.scanner.service.ScannerServiceApi;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.HashSet;
+import java.util.List;
+
+/**
+ * Tests that the mock is compatible with the library's instantiation methods: LaserScanner and ScannerService.
+ */
+public class MockTest {
+
+    private static final Context ctx = InstrumentationRegistry.getContext();
+
+    @Test(timeout = 200) // Have to wait for provider search.
```

Comment:
Ideally never do an arbitrary wait - this often results in random test failures in different environments. If yo uneed to wait for something asynchronous, ideally use a Semaphore that receives a token when ready (and add a waitOnMySemaphore() method that is called by the test)

Reply:
Done, now using semaphores that release permits only once async operations are complete.